### PR TITLE
Configuration::runGitCommand() now respects base-path

### LINKED ISSUE #465: Configuration::runGitCommand() should respect base-dir
## Bug report

| Question    | Answer
| ------------| ---------------
| Box version | 3.8.4
| PHP version | all
| Platform with version | all

I have the following setup:
1. we create the `box.json` dynamically in CI in `/tmp/<unique build directory>` - eg. `/tmp/build-5eeee25/box.json`
2. We build in working directory of code eg. `/build/project-0`

We also want to include the commit hash as placeholder, hovever, Box does run the git commands in the directory of the config instead of running them in the configured `base-dir`.

I am willing to provide a PR but am unsure of how to approach this issue.
IMO it should be implemented here:
https://github.com/humbug/box/blob/6a148c68279a4f0b679b1f2c7e5068b27a7048ec/src/Configuration/Configuration.php#L2719
Currently I am thinking of something like this:
 ```diff
-    private static function runGitCommand(string $command, string $file) : string
+    private static function runGitCommand(string $command, string $path): string
     {
-        $path = dirname($file);
```
Additionally, the path will be added as additional parameter to the  `Configuration::retrieveReplacements()` and all `Configuration::retrieveGit*()` handlers instead of the file.
The value of the path will get determined in `Configuration::create()` as `base-dir` > `dirname($file)` > `getcwd()`.

Does anyone see a problem with this approach?
Am I on the right track that the priorities should be:
1. configured `base-dir` in config
2. base dir of config file.
3. cwd of process

<details>
 <summary>generated `/tmp/build-5eeee25/box.json`</summary>

  ```json
{
    "algorithm": "SHA512",
    "alias": "phpcq",
    "banner": "This file is part of some project",
    "check-requirements": false,
    "compactors": [
        "KevinGH\\\\Box\\\\Compactor\\\\Php",
        "KevinGH\\\\Box\\\\Compactor\\\\PhpScoper",
        "KevinGH\\\\Box\\\\Compactor\\\\Json"
    ],
    "compression": "GZ",
    "datetime": "release-date",
    "git-version": "git-version",
    "base-path": "/build/project-0"
}
 ```
</details>

<details>
 <summary>Output</summary>
 
 ```bash
 $ command
 > output
    ____
    / __ )____  _  __
    / __  / __ \| |/_/
    / /_/ / /_/ />  <
    /_____/\____/_/|_|
    
    
    Box version 3.8.4@120b0a3 2019-12-13 17:22:43 UTC
    
    // Loading the configuration file
    // "/tmp/build-5eeee25/box.json".
    
    
    In Configuration.php line 1153:
    
    The tag or commit hash could not be retrieved from "/tmp/build-5eeee25": The tag or commit hash could not be retrieved from "/tmp/build-5eeee25": fatal: not a git repository (or any of the parent directories): .git
    
    
    In Configuration.php line 1351:
    
    The tag or commit hash could not be retrieved from "/tmp/build-5eeee25": fatal: not a git repository (or any of the parent directories): .git
    
    
    compile [-c|--config CONFIG] [--debug] [--no-parallel] [--no-restart] [--dev] [--no-config] [--with-docker] [-d|--working-dir WORKING-DIR]
 ```
</details>

### PR DESCRIPTION
**Pull request by discordier**
This resolves a problem when having the configuration file not within the repository itself (eg. in CI environments).

Now the configuration is using the `base-path` for running all git commands.

Fixes #465

### CODE REVIEW

**Comment by discordier** [comment]
Failing tests appear to be unrelated to my changes - I have no clue how to address them.

**Comment by theofidry** [comment]
Thanks, will try to check this weekend!

**Comment by discordier** [comment]
Is there anything I can do to assist?

**Comment by theofidry** [comment]
Sorry I appear to have forgotten to submit the review...

I'm away this week, once back I'll have a look at #468 and hopefully this PR will be green and ready to merge :)

File: tests/Configuration/ConfigurationTest.php [hunk 1]
```diff
@@ -1431,6 +1431,96 @@ public function test_the_replacement_map_can_be_configured(): void
...
+
+        $this->setConfig([
+            'base-path' => $basePath,
+            'files' => [self::DEFAULT_FILE],
+            'git' => 'git',
+            'git-commit' => 'git_commit',
+            'git-commit-short' => 'git_commit_short',
+            'git-tag' => 'git_tag',
+            'git-version' => 'git_version',
+            'replacements' => ['rand' => $rand = random_int(0, getrandmax())],
```
[the remark this answers is not in the document]
Comment (discordier) [reply tests/Configuration/ConfigurationTest.php]: Changed with 4aa9f90

**Comment by discordier** [comment]
Rebased master, hopefully CI is green now. :fearful:

**Comment by theofidry** [comment]
It does :) Thanks very much @discordier

**Comment by discordier** [comment]
Thanks for merging.

**Pull request merged**